Modules/core/Controller/CorespaceaccessController.php

```diff
@@ -349,4 +343,59 @@ public function pendingusereditAction($id_space, $id) {
         ));
     }
 
+    /**
+     * 
+     * Reject a pending user
+     * (sets validate=0 && validated_by=<logged user id> in core_pending_accounts)
+     * 
+     * @param int $id_space
+     * @param int $id_pendingAccount
+     */
+    public function pendinguserdeleteAction($id_space, $id_pendingAccount) {
+        $this->checkAuthorization(CoreStatus::$ADMIN);
+        $modelPending = new CorePendingAccount();
+        $modelPending->invalidate($id_pendingAccount, $_SESSION["id_user"]);
+        $mailParams = ["id_space" => $id_space, "id_user" => $modelPending->get($id_pendingAccount)["id_user"]];
+        $this->notifyUserByEmail($mailParams, "reject_pending_user");
+        $this->redirect("corespacependingusers/".$id_space);
+    }
+
+    /**
+     * 
+     * Send an Email to user. Manage the following cases :
+     * - user account has been created
+     * - user is accepted as a member of space
+     * - user request to join a space is rejected
+     *
+     * @param array $params required to fill sendEmail() parameters. Depends on why we want to notify user
+     * @param string $origin determines how to get sendEmail() paramters from $params
+     */
+    public function notifyUserByEmail($params, $origin) {
+        Configuration::getLogger()->debug("notifyUserByEmail", ["params" => $params]);
+        $lang = $this->getLanguage();
+        $fromName = "Platform-Manager";
+        $from = Configuration::get('smtp_from');
+        $spaceName = $params["space_name"];
+
+        if ($origin === "add_new_user_to_space") {
+            $fromName = "Platform-Manager";
+            $toAdress = $params["email"];
+            $subject = CoreTranslator::AccountCreatedSubject($lang, $spaceName);
+            $content = CoreTranslator::AccountCreatedEmail($lang, $params["login"], $params["pwd"]);
+        } else if ($origin === "accept_pending_user" || $origin === "reject_pending_user") {
+            $accepted = ($origin === "accept_pending_user") ? true : false;
+            $userModel = new CoreUser();
+            $pendingUser = $userModel->getInfo($params["id_user"]);
+            $userFullName = $pendingUser["firstname"] . " " . $pendingUser["name"];
+            $subject = CoreTranslator::JoinResponseSubject($spaceName, $lang);
```

Comment:
to be coherent, params should be in same order

CoreTranslator::AccountCreatedSubject($lang, $spaceName);
vs
CoreTranslator::JoinResponseSubject($spaceName, $lang);